api/v1alpha1/edgedeployment_webhook.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+//+kubebuilder:docs-gen:collapse=Go imports
+
+/*
+Next, we'll setup a logger for the webhooks.
+*/
+
+var edgedeploymentlog = logf.Log.WithName("edgedeployment-resource")
+
+/*
+Then, we set up the webhook with the manager.
+*/
+
+func (r *EdgeDeployment) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+/*
+This marker is responsible for generating a validating webhook manifest.
+*/
+
+//+kubebuilder:webhook:verbs=create;update;delete,path=/validate-management-k4e-io-v1alpha1-edgedeployment,mutating=false,failurePolicy=fail,groups=management.k4e.io,resources=edgedeployments,versions=v1alpha1,name=edgedeploymnet.management.k4e.io,sideEffects=None,admissionReviewVersions=v1
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *EdgeDeployment) ValidateCreate() error {
+	edgedeploymentlog.Info("validate create", "name", r.Name)
+
+	return r.validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *EdgeDeployment) ValidateUpdate(old runtime.Object) error {
+	edgedeploymentlog.Info("validate update", "name", r.Name)
+
+	return r.validate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *EdgeDeployment) ValidateDelete() error {
+	edgedeploymentlog.Info("validate delete", "name", r.Name)
+
+	return nil
+}
+
+func (r *EdgeDeployment) validate() error {
+	var notValidPaths []string
```

Comment:
I think that here instead of using an array of strings for errors, I would use go-multierror something that is already used in multiple places.

Reply:
I don't want 'validate' to determine how the error is produced. it only reports 'bad' paths. just like a parser (e.g. XML, JSON and so on). we may want to reuse the function.

Reply:
the function returns an error. how the error is created is an internal matter. besides, GO gives me all the tools I need. and multierror is not appropriate. it is 'multi' and 'error'. here we have multi string.

Reply:
+1